common/core/gpudevice.cpp

```diff
@@ -652,39 +647,44 @@ void GpuDevice::InitializeHotPlugEvents(bool take_lock) {
 }
 
 void GpuDevice::HandleRoutine() {
-  thread_sync_lock_.lock();
-  HandleHWCSettings();
-
-  // Iniitialize resources to monitor external events.
-  // These can be two types:
-  // 1) We are showing splash screen and another App
-  //    needs to take the control. In this case splash
-  //    is true.
-  // 2) Another app is having control of display and we
-  //    we need to take control.
-  // TODO: Add splash screen support.
-  lock_fd_ = open("/vendor/hwc.lock", O_RDONLY);
-  if (lock_fd_ == -1) {
-    thread_sync_lock_.unlock();
-    return;
-  }
-
-  thread_sync_lock_.unlock();
-  display_manager_->IgnoreUpdates();
+  // do nothing.
+}
 
-  if (flock(lock_fd_, LOCK_EX) != 0)
-    ETRACE("Failed to wait on hwc lock.");
+void GpuDevice::HandleWait() {
 
-  close(lock_fd_);
-  lock_fd_ = -1;
+  ETRACE("Called in GpuDevice::HandleWait.");
 
-  display_manager_->ForceRefresh();
-}
+  if (use_lock_file && -1 == lock_fd_) {
+    lock_fd_ = open("/vendor/hwc.lock", O_RDONLY);
 
-void GpuDevice::HandleWait() {
-  if (lock_fd_ == -1) {
-    HWCThread::HandleWait();
+    if (-1 == lock_fd_) {
+      ETRACE("/vendor/hwc.lock doesn't exist, skip the DRM locking");
+      use_lock_file = false;
```

Comment:
hmm in this case where we fail to lock the file, the thread will always exist even if sleeping. That's why we should still try it in the main thread like in the original code and abort the thread in case we are not able to open this hwc.lock file.